src/index.ts

```diff
@@ -14,6 +14,601 @@
+   * Use {@link PreciseDate#getFullTimeString} to get the time as a string.
+   *
+   * @see {@link https://github.com/tc39/proposal-bigint|BigInt}
+   *
+   * @throws {error} If `BigInt` is unavailable.
+   * @returns {bigint}
+   *
+   * @example
+   * const date = new PreciseDate('2019-02-08T10:34:29.481145231Z');
+   *
+   * console.log(date.getFullTime());
+   * // expected output: 1549622069481145231n
+   */
+  getFullTime(): bigint {
+    if (typeof BigInt !== 'function') {
+      throw new Error(NO_BIG_INT);
+    }
+
+    return BigInt(this.getFullTimeString());
+  }
+  /**
+   * Returns a string of the specified date represented in nanoseconds according
+   * to universal time.
+   *
+   * @returns {string}
+   *
+   * @example
+   * const date = new PreciseDate('2019-02-08T10:34:29.481145231Z');
+   *
+   * console.log(date.getFullTimeString());
+   * // expected output: "1549622069481145231"
+   */
+  getFullTimeString(): string {
+    const seconds = this._getSeconds();
+    let nanos = this._getNanos();
+
+    if (nanos && Math.sign(seconds) === Sign.NEGATIVE) {
+      nanos = 1e9 - nanos;
+    }
+
+    return `${seconds}${padLeft(nanos, 9)}`;
+  }
+  /**
+   * Returns the microseconds in the specified date according to universal time.
+   *
+   * @returns {number}
+   *
+   * @example
+   * const date = new PreciseDate('2019-02-08T10:34:29.481145Z');
+   *
+   * console.log(date.getMicroseconds());
+   * // expected output: 145
+   */
+  getMicroseconds(): number {
+    return this._micros;
+  }
+  /**
+   * Returns the nanoseconds in the specified date according to universal time.
+   *
+   * @returns {number}
+   *
+   * @example
+   * const date = new PreciseDate('2019-02-08T10:34:29.481145231Z');
+   *
+   * console.log(date.getNanoseconds());
+   * // expected output: 231
+   */
+  getNanoseconds(): number {
+    return this._nanos;
+  }
+  /**
+   * Sets the microseconds for a specified date according to universal time.
+   *
+   * @param {number} microseconds A number representing the microseconds.
+   * @returns {string} Returns a string representing the nanoseconds in the
+   *     specified date according to universal time.
+   *
+   * @example
+   * const date = new PreciseDate();
+   *
+   * date.setMicroseconds(149);
+   *
+   * console.log(date.getMicroseconds());
+   * // expected output: 149
+   */
+  setMicroseconds(micros: number): string {
+    const abs = Math.abs(micros);
+    let millis = this.getUTCMilliseconds();
+
+    if (abs >= 1000) {
+      millis += Math.floor(abs / 1000) * Math.sign(micros);
+      micros %= 1000;
+    }
+
+    if (Math.sign(micros) === Sign.NEGATIVE) {
+      millis -= 1;
+      micros += 1000;
+    }
+
+    this._micros = micros;
+    this.setUTCMilliseconds(millis);
+
+    return this.getFullTimeString();
+  }
+  /**
+   * Sets the nanoseconds for a specified date according to universal time.
+   *
+   * @param {number} nanoseconds A number representing the microseconds.
```

Comment:
Should this say "representing the nanoseconds"?

Reply:
Whoops, copypasta fail